src/Symfony/Bundle/FrameworkBundle/Command/ContainerDebugCommand.php

```diff
@@ -176,19 +179,28 @@ protected function getContainerBuilder()
             return $this->containerBuilder;
         }
 
-        if (!$this->getApplication()->getKernel()->isDebug()) {
+        $kernel = $this->getApplication()->getKernel();
+
+        if (!$kernel->isDebug()) {
```

Comment:
do we actually need to keep this restriction ? Previously, it was because you the compiler pass was only dumping the file in debug mode. But you don't rely on this file anymore.